start-db-calcite/src/main/java/org/urbcomp/start/db/function/coordtransform/AbstractCoordTransformer.java

```diff
@@ -0,0 +1,117 @@
+    }
+
+    public Polygon polygonTransform(Polygon polygon) {
+        GeometryFactory geometryFactory = GeometryFactoryUtils.defaultGeometryFactory();
+        LinearRing shell = geometryFactory.createLinearRing(
+            lineStringTransform(polygon.getExteriorRing()).getCoordinateSequence()
+        );
+        LinearRing[] holes = new LinearRing[polygon.getNumInteriorRing()];
+        for (int i = 0; i < polygon.getNumInteriorRing(); i++) {
+            LinearRing linearRing = polygon.getInteriorRingN(i);
+            LineString lineString = lineStringTransform(linearRing);
+            holes[i] = geometryFactory.createLinearRing(lineString.getCoordinateSequence());
+        }
+        return geometryFactory.createPolygon(shell, holes);
+    }
+
+    public MultiLineString multiLineStringTransform(MultiLineString multiLineString) {
+        GeometryFactory geometryFactory = GeometryFactoryUtils.defaultGeometryFactory();
+        LineString[] lineStrings = new LineString[multiLineString.getNumGeometries()];
+        for (int i = 0; i < multiLineString.getNumGeometries(); i++) {
+            LineString lineString = (LineString) multiLineString.getGeometryN(i);
+            lineStrings[i] = lineStringTransform(lineString);
+        }
+        return geometryFactory.createMultiLineString(lineStrings);
+    }
+
+    public MultiPolygon multiPolygonTransform(MultiPolygon mPolygon) {
+        GeometryFactory geometryFactory = GeometryFactoryUtils.defaultGeometryFactory();
+        Polygon[] polygons = new Polygon[mPolygon.getNumGeometries()];
+        for (int i = 0; i < mPolygon.getNumGeometries(); i++) {
+            Polygon polygon = (Polygon) mPolygon.getGeometryN(i);
+            polygons[i] = polygonTransform(polygon);
+        }
+        return geometryFactory.createMultiPolygon(polygons);
+    }
+
+    public Trajectory trajectoryTransform(Trajectory trajectory) {
+        List<GPSPoint> points = trajectory.getGPSPointList().stream().map(o -> {
+            double[] coords = transform(o.getLng(), o.getLat());
+            return new GPSPoint(o.getTime(), coords[0], coords[1]);
+        }).collect(Collectors.toList());
+        return new Trajectory(trajectory.getTid(), trajectory.getOid(), points);
+    }
+
+    public RoadSegment roadSegmentTransform(RoadSegment rs) {
+        List<SpatialPoint> points = rs.getPoints().stream().map(o -> {
+            double[] coords = transform(o.getLng(), o.getLat());
+            return new SpatialPoint(coords[0], coords[1]);
+        }).collect(Collectors.toList());
+        return new RoadSegment(
+            rs.getRoadSegmentId(),
+            rs.getStartNode().getNodeId(),
+            rs.getEndNode().getNodeId(),
+            points
+        );
+    }
+
+    public RoadNetwork roadNetworkTransform(RoadNetwork rn) {
+        List<RoadSegment> roadSegments = new ArrayList<>();
+        rn.getRoadSegments().forEach(o -> roadSegments.add(roadSegmentTransform(o)));
```

Comment:
List<RoadSegment> roadSegments = rn.getRoadSegments().stream().map(this::roadSegmentTransform).collect(Collectors.toList());